scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,232 @@
+     * @param fileNameRegexString filename pattern regex which should not be present
+     * @param errorToShow error to show incase of failure
+     * @return [Boolean] check failed or passed
+     */
+    fun checkProhibitedFileNamePattern(
+      repoPath: String,
+      searchFiles: Sequence<File>,
+      prohibitedFilenameRegexString: String,
+      errorToShow: String
+    ): Boolean {
+      var filenamePatternCheckFailedFlag = false
+      val prohibitedFilenameRegex = prohibitedFilenameRegexString.toRegex()
+      searchFiles.forEach {
+        val filePath = it.toString().removePrefix(repoPath)
+        if (prohibitedFilenameRegex.matches(filePath)) {
+          logProhibitedFilenameFailure(
+            filePath = filePath,
+            errorToShow = errorToShow
+          )
+          filenamePatternCheckFailedFlag = true
+        }
+      }
+      return filenamePatternCheckFailedFlag
+    }
+
+    /**
+     * Checks for a prohibited file content
+     *
+     * @param repoPath the path of the repo.
+     * @param searchFiles a list of all the files which needs to be checked.
+     * @param fileNameRegexString filename regex string in which to do the content check
+     * @param prohibitedContentRegexString regex string which should not be contained in the file
+     * @param errorToShow error to show incase of failure
+     * @return [Boolean] check failed or passed
+     */
+    fun checkProhibitedContent(
+      repoPath: String,
+      searchFiles: Sequence<File>,
+      fileNameRegexString: String,
+      prohibitedContentRegexString: String,
+      errorToShow: String
+    ): Boolean {
+      var contentCheckFailedFlag = false
+      val fileNameRegex = fileNameRegexString.toRegex()
+      val prohibitedContentRegex = prohibitedContentRegexString.toRegex()
+      searchFiles.forEach {
+        if (fileNameRegex.matches(it.name)) {
+          var lineNumber = 0
+          File(it.toString()).forEachLine { lineString ->
+            lineNumber++
+            if (prohibitedContentRegex.matches(lineString)) {
+              logProhibitedContentFailure(
+                lineNumber = lineNumber,
+                errorToShow = errorToShow,
+                filePath = it.toString().removePrefix(repoPath)
+              )
+              contentCheckFailedFlag = true
+            }
+          }
+        }
+      }
+      return contentCheckFailedFlag
+    }
+
+    /**
+     * Collects the paths of all the files which are needed to be checked
+     *
+     * @param repoPath the path of the repo.
+     * @param allowedDirectories a list of all the directories which needs to be checked.
+     * @param exemptionList a list of files which needs to be exempted for this check
+     * @return [Sequence<File>] all files which needs to be checked.
+     */
+    fun collectSearchFiles(
+      repoPath: String,
+      allowedDirectories: MutableList<String>,
+      exemptionList: Array<String> = arrayOf<String>()
```

Comment:
Shouldn't exemptions be per-check? In that case, we should be defining them as part of the proto/textproto file. Otherwise, we'd be exempting files for every regex check which doesn't sound correct to me.

The per-regex check exemption seems nice, though, since we'd use this in the future for new checks added.

Reply:
Revamped to be based on a `per-regex` approach. PTAL.